python/sidecar/search.py

```diff
@@ -0,0 +1,256 @@
+                "not",
+                "only",
+                "own",
+                "same",
+                "so",
+                "than",
+                "too",
+                "very",
+                "s",
+                "t",
+                "can",
+                "will",
+                "just",
+                "don",
+                "don't",
+                "should",
+                "should've",
+                "now",
+                "d",
+                "ll",
+                "m",
+                "o",
+                "re",
+                "ve",
+                "y",
+                "ain",
+                "aren",
+                "aren't",
+                "couldn",
+                "couldn't",
+                "didn",
+                "didn't",
+                "doesn",
+                "doesn't",
+                "hadn",
+                "hadn't",
+                "hasn",
+                "hasn't",
+                "haven",
+                "haven't",
+                "isn",
+                "isn't",
+                "ma",
+                "mightn",
+                "mightn't",
+                "mustn",
+                "mustn't",
+                "needn",
+                "needn't",
+                "shan",
+                "shan't",
+                "shouldn",
+                "shouldn't",
+                "wasn",
+                "wasn't",
+                "weren",
+                "weren't",
+                "won",
+                "won't",
+                "wouldn",
+                "wouldn't",
+            ]
+        )
+
+    # Method to preprocess the query
+    def preprocess_query(self, query):
+        # convert query to lowercase and remove stopwords
+        query = query.lower()
+        query = "+".join([word for word in query.split() if word not in self.stopwords])
+        return query
+
+    # Method to perform the search
+    # Other search fields can be found here:
+    # https://github.com/danielnsilva/semanticscholar/blob/da433a1f45e0c17bbd58300de0602c969602675a/semanticscholar/Paper.py#L93
+    def search(
+        self,
+        query,
+        limit=10,
+        search_fields=[
+            "title",
+            "abstract",
+            "venue",
+            "year",
+            "corpusId",
+            "citationCount",
+            "openAccessPdf",
+        ],
+    ):
+        # Preprocess the query
+        query = self.preprocess_query(query)
+        results = self.s2.search_paper(query, limit=limit, fields=search_fields)
+        results_list = []
+        # For each result, create a dictionary with desired fields
+        for item in results[:limit]:
+            result = {field: getattr(item, field, None) for field in search_fields}
+            results_list.append(result)
+        return results_list
+
+
+# Main function to parse arguments and perform the search
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Semantic Scholar search")
```

Comment:
The parser is configured in the ``cli.py` file and not here.

Reply:
changes made here - 8e3fdf39e92caad43c3091887acfd2d518e00b25